airflow-core/src/airflow/cli/commands/config_command.py

```diff
@@ -704,3 +705,148 @@ def lint_config(args) -> None:
+            airflow config update --verbose
+
+    :param args: The CLI arguments for updating configuration.
+    """
+    console = AirflowConsole()
+    changes_applied = []
+
+    update_sections = args.section if args.section else None
+    update_options = args.option if args.option else None
+    ignore_sections = args.ignore_section if args.ignore_section else []
+    ignore_options = args.ignore_option if args.ignore_option else []
+
+    for change in CONFIGS_CHANGES:
+        conf_section = change.config.section
+        conf_option = change.config.option
+
+        if update_sections is not None and conf_section not in update_sections:
+            continue
+        if update_options is not None and conf_option not in update_options:
+            continue
+        if conf_section in ignore_sections or conf_option in ignore_options:
+            continue
+
+        if not conf.has_option(conf_section, conf_option, lookup_from_deprecated=False):
+            continue
+
+        current_value = conf.get(conf_section, conf_option)
+
+        if change.default_change and (str(current_value) != str(change.new_default)):
+            conf.set(conf_section, conf_option, str(change.new_default))
+            changes_applied.append(
+                f"Updated default value of '{conf_section}/{conf_option}' from '{current_value}' to '{change.new_default}'."
+            )
+
+        if change.renamed_to:
+            new_section = change.renamed_to.section
+            new_option = change.renamed_to.option
+            if not conf.has_section(new_section):
+                conf.add_section(new_section)
+            conf.set(new_section, new_option, current_value)
+            conf.remove_option(conf_section, conf_option)
+            changes_applied.append(f"Renamed '{conf_section}/{conf_option}' to '{new_section}/{new_option}'.")
+        elif change.was_removed:
+            conf.remove_option(conf_section, conf_option)
+            changes_applied.append(f"Removed '{conf_section}/{conf_option}' from configuration.")
+
+    for change in CONFIGS_CHANGES:
+        conf_section = change.config.section
+        conf_option = change.config.option
+
+        if update_sections is not None and conf_section not in update_sections:
+            continue
+        if update_options is not None and conf_option not in update_options:
+            continue
+        if conf_section in ignore_sections or conf_option in ignore_options:
+            continue
+
+        old_env = f"{ENV_VAR_PREFIX}{conf_section.upper()}__{conf_option.upper()}"
+        if old_env in os.environ:
+            os.environ.pop(old_env)
+            if change.renamed_to:
+                new_env = (
+                    f"{ENV_VAR_PREFIX}{change.renamed_to.section.upper()}__{change.renamed_to.option.upper()}"
+                )
+                new_value = conf.get(change.renamed_to.section, change.renamed_to.option)
+                os.environ[new_env] = new_value
```

Comment:
Same here, this isn't persistent.

Reply:
Removed checking env completely and keeping this command to handle airflow.cfg.